components/raftstore-v2/src/operation/command/admin/commit_merge.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 TiKV Project Authors. Licensed under Apache-2.0.
+
+use kvproto::{
+    raft_cmdpb::{AdminCmdType, RaftCmdRequest},
+    raft_serverpb::{MergeState, PeerState, RegionLocalState},
+};
+use slog::{info, Logger};
+
+#[derive(Default)]
+pub struct MergeContext {
+    /// Used at source region ///
+    /// When a fence <idx, cmd> is present, we (1) delay the PrepareMerge
+    /// command `cmd` until all writes before `idx` are applied (2) reject all
+    /// in-coming write proposals.
+    /// Before proposing `PrepareMerge`, we first serialize and propose the lock
+    /// table. Locks marked as deleted (but not removed yet) will be
+    /// serialized as normal locks.
+    /// Thanks to the fence, we can ensure at the time of lock transfer, locks
+    /// are either removed (when applying logs) or won't be removed before
+    /// merge (the proposals to remove them are rejected).
+    pub prepare_fence: Option<u64>,
+    /// Used when `prepare_fence` is present.
+    pub pending_prepare: Option<RaftCmdRequest>,
```

Comment:
How about making them in `Enum`? As `pending_prepare` and `pending` won't show up at the same time.